src/java/com/zimbra/oauth/utilities/LdapConfiguration.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra OAuth Social Extension
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.oauth.utilities;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+
+/**
+ * The Configuration class for this project for loading attributes from LDAP
+ * @author zimbra
+ *
+ */
+public class LdapConfiguration extends Configuration {
+
+    
+    /**
+     * @param appName
+     * @throws ServiceException
+     */
+    protected LdapConfiguration(String appName) {
+        super(appName);    
+    }
+    
+    
+    /**
+     * 
+     */
+    @Override
+    public String getString(String key) {
+        
+        String appName = getAppName(key);
+        return getConfig(key, appName);
+    }
+
+    /**
+     * @param oauthClientId
```

Comment:
A brief description of, method, params & return.

Reply:
Done